website/content/v0.14/guides/configuring-ceph-with-rook.md

```diff
@@ -0,0 +1,278 @@
+
+```shell
+$ kubectl --namespace rook-ceph get cephclusters.ceph.rook.io rook-ceph
+NAME        DATADIRHOSTPATH   MONCOUNT   AGE   PHASE   MESSAGE                        HEALTH      EXTERNAL
+rook-ceph   /var/lib/rook     3          98m   Ready   Cluster created successfully   HEALTH_OK
+```
+
+If it is, you can begin the upgrade process for the Talos Linux node, during which time the Ceph cluster will become unhealthy as the node is reconfigured.
+Before performing any other action on the Talos Linux nodes, the Ceph cluster must return to a healthy status.
+
+```shell
+$ talosctl upgrade --nodes 172.20.15.5 --image ghcr.io/talos-systems/installer:v0.14.3
+NODE          ACK                        STARTED
+172.20.15.5   Upgrade request received   2022-03-27 20:29:55.292432887 +0200 CEST m=+10.050399758
+
+$ kubectl --namespace rook-ceph get cephclusters.ceph.rook.io
+NAME        DATADIRHOSTPATH   MONCOUNT   AGE   PHASE         MESSAGE                   HEALTH        EXTERNAL
+rook-ceph   /var/lib/rook     3          99m   Progressing   Configuring Ceph Mgr(s)   HEALTH_WARN
+
+$ kubectl --namespace rook-ceph wait --timeout=1800s --for=jsonpath='{.status.ceph.health}=HEALTH_OK' rook-ceph
+cephcluster.ceph.rook.io/rook-ceph condition met
+```
+
+The above steps need to be performed for each Talos Linux node undergoing maintainence, one at a time.
+
+## Cleaning Up
+
+### Rook Ceph Cluster Removal
+
+Removing a Rook Ceph cluster requires a few steps, starting with signalling to Rook that the Ceph cluster is really being destroyed.
+Then all Persistent Volumes (and Claims) backed by the Ceph cluster must be deleted, followed by the Storage Classes and the Ceph storage types.
+
+```shell
+$ kubectl --namespace rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
+cephcluster.ceph.rook.io/rook-ceph patched
+
+$ kubectl delete storageclasses ceph-block ceph-bucket ceph-filesystem
+storageclass.storage.k8s.io "ceph-block" deleted
+storageclass.storage.k8s.io "ceph-bucket" deleted
+storageclass.storage.k8s.io "ceph-filesystem" deleted
+
+$ kubectl --namespace rook-ceph delete cephblockpools ceph-blockpool
+cephblockpool.ceph.rook.io "ceph-blockpool" deleted
+
+$ kubectl --namespace rook-ceph delete cephobjectstore ceph-objectstore
+cephobjectstore.ceph.rook.io "ceph-objectstore" deleted
+
+$ kubectl --namespace rook-ceph delete cephfilesystem ceph-filesystem
+cephfilesystem.ceph.rook.io "ceph-filesystem" deleted
+```
+
+Once that is complete, the Ceph cluster itself can be removed, along with the Rook Ceph cluster Helm chart installation.
+
+```shell
+$ kubectl --namespace rook-ceph delete cephcluster rook-ceph
+cephcluster.ceph.rook.io "rook-ceph" deleted
+
+$ helm --namespace rook-ceph uninstall rook-ceph-cluster
+release "rook-ceph-cluster" uninstalled
+```
+
+If needed, the Rook Operator can also be removed along with all the Custom Resource Definitions that it created.
```

Comment:
probably add a note to wait for the rook operator to do all cleanup process before deleting the rook-operator

Reply:
ohh nice :muscle: , I've some operators not doing finalizers properly and having the operator deleted before the cleanup happened.